clients/flash/common-library/src/main/actionscript/org/bigbluebutton/lib/chat/views/ChatViewMediatorBase.as

```diff
@@ -88,6 +100,29 @@ package org.bigbluebutton.lib.chat.views {
 			}
 		}
 		
+		private function onLockSettingsChanged(newSettings:LockSettings2x):void {
+			applyLockSettings(newSettings);
+		}
+		
+		private function applyLockSettings(lockSettings:LockSettings2x):void {
+			// Lock settings applies only to viewers.
+			if (meetingData.users.me.role == UserRole.MODERATOR) return;

```

Comment:
if (MODERATOR || !locked) return

Reply:
This functions handles locking and unlocking. If we add !locked, then unlocking won't get applied.

Reply:
Very good point. That actually raises another issue. If the user's moderator status changes it won't update either.

Reply:
We'll need to listen for the role status changed and re-apply lock settings.